tests/k8s_handler.py

```diff
@@ -198,3 +199,34 @@ def wait_for_namespaced_deletion(name, namespace, deletion_func, list_func, time
             if item.metadata.name == name:
                 deleted = False
     return deleted
+
+
+def wait_for_daemonset_to_be_ready(name, namespace, timeout=None):
+    wait_for_to_be_ready("daemonset", name, namespace, timeout=timeout)
+
+
+def resolve_read_status_func(obj_name):
+    if obj_name == "daemonset":
+        return client.AppsV1Api().read_namespaced_daemon_set_status
+    else:
+        raise ValueError(f"resolve_read_status_func: {obj_name} is not a valid value")
```

Comment:
if this error occurs it will crash the entire run - is this function something we'd like to retry?

Reply:
it's a setup for other waiting functions to be included here (deployment and statefulset) though I have other tasks to attend to at the moment so I can't really .
right now this function is only utilised for daemonset and that's the only value to be given here so no worries about exceptions at this point.
we already now failing tests if pods are not ready in time so it seemed right to throw an exception here but that might be changed once I change this func to include other waiting functions as well.